### PR TITLE
fix: checkstyle's configFile was erroneous

### LINKED ISSUE #14: Checkstyle's configPath is erroneous
Hi there!

Removing the old checkstyle's config from my project made the build failing using the version `0.0.2-SNAPSHOT` of the plugin.

**What happened**

The `checkstyleIntegrationTest` was failing after the old configuration has been removed.
```
> Task :checkstyleIntegrationTest FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':checkstyleIntegrationTest'.
> Unable to create Root Module
```
Checking what's going on with `--stacktrace` shown me the underlying cause:

`Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Unable to find: /home/fthouraud/Development/ekinow/api/gateway/file:/home/fthouraud/.gradle/caches/modules-2/files-2.1/com.ekino.oss.gradle.plugin/gradle-quality-plugin/0.0.2-SNAPSHOT/783f43c13cb521a9c535454a4d9804d8c6ec43be/gradle-quality-plugin-0.0.2-SNAPSHOT.jar!/checkstyle.xml`

The path of the resource has been prefixed by the absolute path of the project which can't work due to the resource path starting with `file` and already being absolute.

**What is expected**

We can expect that the checkstyle task could load its configuration from the plugin and run the analysis like intended.

It seems to work as intented before the `0.0.2-SNAPSHOT` but I can't tell which modification introduced this.

**Analysis**

I thought at first that was somewhat related to this [issue](https://github.com/gradle/gradle/issues/8286) but checking with both Java 8 and 11 gave me the same results.

To check this, I created a task to print some informations:
```
> Task :bug
*********************
Resource path: file:/home/fthouraud/.gradle/caches/modules-2/files-2.1/com.ekino.oss.gradle.plugin/gradle-quality-plugin/0.0.2-SNAPSHOT/783f43c13cb521a9c535454a4d9804d8c6ec43be/gradle-quality-plugin-0.0.2-SNAPSHOT.jar!/checkstyle.xml
As file: file:/home/fthouraud/.gradle/caches/modules-2/files-2.1/com.ekino.oss.gradle.plugin/gradle-quality-plugin/0.0.2-SNAPSHOT/783f43c13cb521a9c535454a4d9804d8c6ec43be/gradle-quality-plugin-0.0.2-SNAPSHOT.jar!/checkstyle.xml

Extension's configFile: /home/fthouraud/Development/ekinow/api/gateway/file:/home/fthouraud/.gradle/caches/modules-2/files-2.1/com.ekino.oss.gradle.plugin/gradle-quality-plugin/0.0.2-SNAPSHOT/783f43c13cb521a9c535454a4d9804d8c6ec43be/gradle-quality-plugin-0.0.2-SNAPSHOT.jar!/checkstyle.xml
*********************
```
This confirm that the path is biased on the `CheckstyleExtension` side.

**Possible solution**

Since Gradle 2.2, there's a `config` property on the `CheckstyleExtension` object which rely on a `TextResource` instead of a `File` ([Gradle 6.0 CheckstyleExtension](https://docs.gradle.org/current/javadoc/org/gradle/api/plugins/quality/CheckstyleExtension.html#setConfig-org.gradle.api.resources.TextResource-)).

We could then replace this part:
```kotlin
configure<CheckstyleExtension> {
  toolVersion = "8.24"
  configFile = getCheckstyleConfig()
}
```
with:
```kotlin
configure<CheckstyleExtension> {
  toolVersion = "8.24"
  config = resources.text.fromFile(getCheckstyleConfig())
}
```

If this seems okay to you I could propose a PR.

Thanks

### PR DESCRIPTION
**Pull request by fthouraud**
fix #14

### CODE REVIEW

**Comment by Crow-EH** [comment]
Seems like unbroken-dome test-sets 2.1.0 is unreachable (thanks @saugereau).
@fthouraud I thing you'll have to upgrade it to fix the build.

**Comment by Crow-EH** [comment]
The tests are still failing.

As we discussed with @fthouraud and @saugereau, it may be due to [Gradle's recent upgrade of TLS from 1.2 to 1.3](https://status.gradle.com/incidents/q2vq49fndxg0). There is an [issue on JDK regarding TLS 1.3](https://bugs.openjdk.java.net/browse/JDK-8213202) (that Gradle links to in the previous link) that has been fixed in 11.0.3.

Sadly, Travis still installs JDK 11.0.2 (see tests logs) due to its use of https://github.com/sormuras/bach/. Bach doesn't want to change its JDK download source (https://github.com/sormuras/bach/pull/89), even though [java.net is clearly only storing old archives of JDK 11](http://jdk.java.net/archive/).

The latest versions of JDK 11's *binaries* seem to only be available on oracle, adopt and other vendors' sites.

# TL;DR
IMHO, we have to either:
* Convince Bach to change its way and contribute (**hardest solution**)
* Switch to a [containerized build on Travis](https://docs.travis-ci.com/user/docker/) (**easiest solution**)
* Switch to ✨GitHub Actions✨ and https://github.com/actions/setup-java (**funniest solution** 😁)

WDYT @fthouraud @saugereau ?

**Comment by fthouraud** [comment]
Thanks @Crow-EH for the summary.

Bach removed the support for Oracle & AdoptOpenJDK so I don't think we could manage to change his mind.
But, there's still an easier solution considering that we can specify a custom URL to `install-jdk.sh` with the `--url` flag (cf [Bach's comment](https://github.com/sormuras/bach/issues/51#issuecomment-487385445)).

I never used GitHub Actions but I won't doubt it would be funnier :smile:

File: .travis/build.sh [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #!/usr/bin/env bash
 
-./gradlew build
+./gradlew build --info --stacktrace
```
Comment (saugereau) [inline .travis/build.sh]: Can you remove --info and --stacktrace ?

**Review by saugereau — CHANGES_REQUESTED** [review]

**Review by Crow-EH — APPROVED** [review]

**Comment by Crow-EH** [comment]
LGTM :+1:

**Review by saugereau — APPROVED** [review]
